src/shared/utils/index.ts

```diff
@@ -357,10 +357,20 @@ export const matchPlugins = (
 ) => {
   const customizer: isMatchWithCustomizer = (value: any, other: any) => {
     if (Array.isArray(other) && !Array.isArray(value)) {
-      return other.length === 1 && other.indexOf(value) >= 0;
+      return other.length === 1 && isMatch(value, other[0]);
     }
+
+    // return true if any object in value array match any object in
+    // other array.
     if (Array.isArray(other) && Array.isArray(value)) {
-      return difference(other, value).length === 0;
+      for (let i = 0; i < value.length; i += 1) {
```

Comment:
it can be just i++ :) classic!

Reply:
Tslint is complaining about ++. https://palantir.github.io/tslint/rules/increment-decrement/
I prefer ++ and -- to be honest.

Reply:
fair enough :)